src/store/auth/sagas.ts

```diff
@@ -69,13 +72,61 @@ import {API_ENDPOINT_MY_INFO, prepare} from 'services/endpoints';
 import {AuthenticationError} from 'services/errors/authentication';
 import {InstanceCheckError} from 'services/errors/instance-check';
 
+/**
+ * Check instance existence & compatibility
+ * @param action this param can be undefined when calling this generator from another generator
+ */
 function* checkInstance(action?: CheckInstanceAction) {
   try {
     yield openLoader();
-    const instanceUrl: string = yield selectInstanceUrl();
-    const response: Response = yield call(checkInstanceRequest, instanceUrl);
+    let instanceUrl: string = yield selectInstanceUrl();
+    let response: Response = yield call(checkInstanceRequest, instanceUrl);
+
+    // check instance in advanced
+    if (!response.ok) {
+      const urlPortions = [
+        '/symfony/web',
+        '/symfony/web/index.php',
+        '/index.php',
+      ];
+      const effects = urlPortions.map((urlPortion) => {
+        return call(checkInstanceRequestWithCatch, instanceUrl + urlPortion);
+      });
+      const responses: Response[] = yield all(effects);
+
+      for (let i = 0; i < responses.length; i++) {
+        if (responses[i]?.ok) {
+          response = responses[i];
+          instanceUrl = instanceUrl + urlPortions[i];
+          break;
+        }
+      }
+    }
+
+    // check instance is legacy
+    if (!response.ok) {
+      const urlPortions = [
```

Comment:
is it necessary define url set two times?  line 97 and line 108

Reply:
done